tests/virt_autotest/reboot_and_wait_up.pm

```diff
@@ -44,7 +44,7 @@ sub reboot_and_wait_up {
         switch_from_ssh_to_sol_console(reset_console_flag => 'off');
         #on some SUTs, fg. a HP machine in Beijing lab, screen of sol console is cleared at the second reboot.
         #and the 'return' key must be typed then 'login' prompt appears in sol console
-        send_key 'ret' unless check_screen('text-logged-in-root');
+        send_key 'ret' if check_screen('sol-console-wait-typing-ret');
```

Comment:
To be honest, I do not think this new needle name is intuitive. I think "sol-console-activated" is better based what you are   capturing. Additionally, "unless check_screen('text-logged-in-root')" already covers "if check_screen('sol-console-wait-typing-ret')", so why bother changing this part ?

Reply:
By the way, I think there is already functionality can help do this, you do not need a new needle:

```
    select_console 'sol', await_console => 1;
    send_key 'ret';
```
This code snippet can help do the same thing.

Reply:
two needles here because I have met the similiar problem you had met where you opened a progress ticket to openqa tool team.

Note the screenshots below:
text logged in: http://10.67.129.51/tests/2573#step/switch_to_ssh_and_install_hypervisor/5
'sol-console-wait-typing-ret'(not type 'ret' yet): http://10.67.129.51/tests/2573#step/switch_to_ssh_and_install_hypervisor/2
sol-console-activated('ret' typed or no need to type 'ret' on some suts):  http://10.67.129.51/tests/2573#step/reboot_and_wait_up_normal/3

Reply:
This is what I think:
```
sub switch_from_ssh_to_sol_console {
    my (%opts) = @_;

    #close root-ssh console
    prepare_system_shutdown;
    #switch to sol console
    set_var('SERIALDEV', '');
    $serialdev = 'ttyS1';
    bmwqemu::save_vars();
    if ($opts{'reset_console_flag'} eq "on") {
        reset_consoles;
    }
    select_console 'sol', await_console => 1; ##### You need to change to await console
    save_screenshot;
}
```
Then remove 
```
send_key 'ret' if check_screen('sol-console-wait-typing-ret');
```

or just
```
send_key 'ret'
```
Because "await_console" already makes sure you are in sol.

Reply:
SUT behaves differently with select_console 'sol' regardless await_console => 1, for example,  sut A(such as suts in OSD) changes to sol console and '#' prompt directly(like this https://openqa.nue.suse.com/tests/6598987#step/reboot_and_wait_up_normal/4); sut B(such as vh015/vh016 HP machines) stops at 'sol-console-wait-typing-ret'(http://10.67.129.51/tests/2573#step/switch_to_ssh_and_install_hypervisor/2),  after receives send_key 'ret' then '#' prompts.

We have to differentiate them because 'ret' is only required for sut B. sut A is prompting login, 'ret' will make login fail on sut A.
The meaning of check_screen('sol-console-wait-typing-ret')  is to distinguish between SUT A and SUT B

here is the example I just run(with 'await_console => 1' and without sending 'ret' on vh016):
http://10.67.129.51/tests/2584#step/switch_to_ssh_and_install_hypervisor/3

Reply:
Why the original
```
send_key 'ret' unless check_screen('text-logged-in-root');
```
is replaced with
```
send_key 'ret' if check_screen('sol-console-wait-typing-ret');
```
Have you considered any consequence that may follow ? There is no chance to "ret' anymore unless "sol-console-wating-type-ret".

Reply:
> Why the original
> 
> ```
> send_key 'ret' unless check_screen('text-logged-in-root');
> ```
> 
> is replaced with
> 
> ```
> send_key 'ret' if check_screen('sol-console-wait-typing-ret');
> ```
> 
'sol-console-wait-typing-ret' is a new introduced needle which can avoid the problem you met in arm. the selected area can be large enough to distinguish 'text-logged-in-root' and 'text-login'.

> Have you considered any consequence that may follow ? There is no chance to "ret' anymore unless 'sol-console-wait-typing-ret"

Yes, that is exactly what I want. no 'ret' for those suts which has already been in sol console, type 'ret' for thost suts which required 'ret' for command prompts.

Reply:
OK if you know what you are going to do.